packages/@sanity/block-tools/src/util/normalizeBlock.ts

```diff
@@ -53,6 +58,12 @@ export default function normalizeBlock(block) {
       if (child._type === 'span' && !child.marks) {
         child.marks = []
       }
+      if (allowedDecorators && child._type === 'span') {
+        child.marks = child.marks.filter(
+          mark =>
+            allowedDecorators.includes(mark) || block.markDefs.map(def => def._key).includes(mark)
```

Comment:
The `map().includes()` check could be written slightly more performant by using  `.some()` or `.find()` - exiting once a match is encountered

Reply:
True!